### PR TITLE
Remove limits from pods

### PR DESCRIPTION
**Pull request by fabianofranz**
Removes CPU limits from pods, soon to be checked by the test in https://github.com/openshift/origin/pull/22095.

Re https://jira.coreos.com/browse/CO-262

### CODE REVIEW

File: config/manager/deployment.yaml [hunk 1]
```diff
@@ -52,9 +52,6 @@ spec:
         imagePullPolicy: Always
         name: manager
         resources:
-          limits:
-            cpu: 100m
-            memory: 500Mi
           requests:
             cpu: 100m
```
Comment (dgoodwin) [inline config/manager/deployment.yaml]: Could you flip the requests to 10m as well, sounded like Derek is going to soon be asking for that as well so we may as well get ready.

File: config/manager/deployment.yaml [hunk 2]
```diff
@@ -52,9 +52,6 @@ spec:
         imagePullPolicy: Always
         name: manager
         resources:
-          limits:
-            cpu: 100m
-            memory: 500Mi
```
Comment (dgoodwin) [inline config/manager/deployment.yaml]: Should the memory limit go as well? His request seemed specific to CPU but I don't know if you can even have a memory limit and no cpu specified.

Comment (fabianofranz) [reply config/manager/deployment.yaml]: The test checks for the `memory` attribute the same way it does the `cpu`, so it seems it needs to go as well.

File: config/manager/deployment.yaml (hunk 1 above)
Comment (fabianofranz) [reply config/manager/deployment.yaml]: What the test does as of now is to check it it's present and if not, it says "recommend *min* value 10m". But I had the same questions. @derekwaynecarr can you check the two comments in this PR and confirm?

File: config/manager/deployment.yaml (hunk 2 above)
Comment (dgoodwin) [reply config/manager/deployment.yaml]: Very good catch, thanks.

Comment (derekwaynecarr) [reply config/manager/deployment.yaml]: i am less concerned about a memory limit than cpu limits.  The test only fails on cpu limits.

File: config/manager/deployment.yaml (hunk 1 above)
Comment (derekwaynecarr) [reply config/manager/deployment.yaml]: a follow on test will validate that we schedule no more than 2 cores of cpu requests on master nodes.  Please set this to 10m.  The cpu time is relative to other components, and 10m is fine.

Comment (derekwaynecarr) [reply config/manager/deployment.yaml]: basically, we will give the api server and etcd equal cpu time by making their request equal ~150m.  we will give the controller manager and scheduler less relative time ~100m.  any operator is basically a single controller in controller manager and giving it 1/10 what we will give controller manager is fine.  The key need is that we give operators matching cpu requests so they have a fair share of cpu time relative to each other, but less than etcd/APIServer

Comment (fabianofranz) [reply config/manager/deployment.yaml]: Nice, thanks for clarifying @derekwaynecarr. PR updated.

**Comment by fabianofranz** [comment]
PR updated to reflect Derek's comments, @dgoodwin PTAL.

**Pull request merged**